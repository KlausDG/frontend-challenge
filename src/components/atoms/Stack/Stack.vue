<script setup lang="ts">
import { computed } from 'vue'

interface Props {
  direction?: 'row' | 'column'
  spacing?: number
}

const props = withDefaults(defineProps<Props>(), {
  direction: 'row',
})

const style = computed(() => ({
  flexDirection: props.direction || 'column',
  gap: props.spacing ? `${props.spacing}px` : '0px',
}))
</script>

<template>
  <div class="stack__container" :style="style">
    <slot />
  </div>
</template>

<style scoped lang="scss">
.stack__container {
  display: flex;
}
</style>
