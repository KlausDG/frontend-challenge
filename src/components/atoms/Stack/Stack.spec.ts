import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import Stack from './Stack.vue'

describe('Stack.vue', () => {
  it('renders default slot content', () => {
    const wrapper = mount(Stack, {
      slots: {
        default: '<div class="slot-content">Slot Content</div>',
      },
    })

    const slotContent = wrapper.find('.slot-content')
    expect(slotContent.exists()).toBe(true)
    expect(slotContent.text()).toBe('Slot Content')
  })

  it('applies row direction by default', () => {
    const wrapper = mount(Stack)
    const container = wrapper.find('.stack__container')
    expect((container.element as HTMLElement).style.flexDirection).toBe('row')
  })

  it('applies column direction when passed via props', () => {
    const wrapper = mount(Stack, {
      props: { direction: 'column' },
    })
    const container = wrapper.find('.stack__container')
    expect((container.element as HTMLElement).style.flexDirection).toBe('column')
  })

  it('applies spacing in px when spacing prop is provided', () => {
    const wrapper = mount(Stack, {
      props: { spacing: 12 },
    })
    const container = wrapper.find('.stack__container')
    expect((container.element as HTMLElement).style.gap).toBe('12px')
  })

  it('defaults spacing to 0px when spacing prop is not provided', () => {
    const wrapper = mount(Stack)
    const container = wrapper.find('.stack__container')
    expect((container.element as HTMLElement).style.gap).toBe('0px')
  })

  it('matches the component snapshot', () => {
    const wrapper = mount(Stack, {
      props: { direction: 'column', spacing: 8 },
      slots: {
        default: '<div>Snapshot Slot</div>',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
