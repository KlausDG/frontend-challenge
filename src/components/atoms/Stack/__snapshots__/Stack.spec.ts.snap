// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stack.vue > matches the component snapshot 1`] = `
"<div data-v-69f42c1a="" class="stack__container" style="flex-direction: column; gap: 8px;">
  <div>Snapshot Slot</div>
</div>"
`;
